serverless/aws/functions/generic.py

```diff
@@ -249,6 +249,12 @@ def with_idempotency(self, table_name=None):
         self.environment = env
 
         return self
+    
+    def add_layer(self, arn, **kwargs):
+        if not getattr(self, "layers", None):
+            setattr(self, "layers", [])
+
+        self.layers.append(arn)
```

Comment:
`self.layers` is not always set by the `__init__` method. I would improve the `__init__` method to always set it.

Reply:
I just realised this is doing setattr on self. I thought I saw it adding it to something else before so only glossed over this part. Anyway, can you do `self.layers = layers or []` in the `__init__` method to avoid adding it here? Or is there a downside to that?

Reply:
I think we dont want layers as an attr of self unless it is populated.   I believe it is because of how the yaml is then generated.  Dont want a definition in the yaml with no values so to speak.

Reply:
Ive realised since my last change, I dont actually need to modify serverless-builder at all.